scripts/gen_quick_start_module.py

```diff
@@ -1,20 +1,152 @@
 #!/usr/bin/env python3
 import json
 import os
-from typing import Dict
+import argparse
+import io
+import sys
+from pathlib import Path
+from typing import Dict, Set, List, Iterable
+from enum import Enum
+
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+class OperatingSystem(Enum):
+    LINUX: str = "linux"
+    WINDOWS: str = "windows"
+    MACOS: str = "macos"
+
+PRE_CXX11_ABI = "pre-cxx11"
+CXX11_ABI = "cxx11-abi"
+DEBUG = "debug"
+RELEASE = "release"
+DEFAULT = "default"
+ENABLE = "enable"
+DISABLE = "disable"
+
+# Mapping json to release matrix is here for now
+# TBD drive the mapping via:
+#  1. Scanning release matrix and picking 2 latest cuda versions and 1 latest rocm
+#  2. Possibility to override the scanning algorithm with arguments passed from workflow
+
+acc_arch_ver_map = {
+    "nightly": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        },
+    "release": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        }
+    }
 
+LIBTORCH_DWNL_INSTR = {
+        PRE_CXX11_ABI: "Download here (Pre-cxx11 ABI):",
+        CXX11_ABI: "Download here (cxx11 ABI):",
+        RELEASE: "Download here (Release version):",
+        DEBUG: "Download here (Debug version):",
+    }
 
 def read_published_versions():
     with open(os.path.join(BASE_DIR, "published_versions.json")) as fp:
         return json.load(fp)
 
+def write_published_versions(versions):
+    with open(os.path.join(BASE_DIR, "published_versions.json"), "w") as outfile:
+            json.dump(versions, outfile, indent=2)
+
+def read_matrix_for_os(osys: OperatingSystem, value: str):
+    try:
+        with open(os.path.join(BASE_DIR, f"{osys.value}_{value}_matrix.json")) as fp:
+            return json.load(fp)["include"]
+    except FileNotFoundError as e:
+        raise ImportError(f"Release matrix not found for: {osys.value} error: {e.strerror}") from e
+
 
 def read_quick_start_module_template():
     with open(os.path.join(BASE_DIR, "_includes", "quick-start-module.js")) as fp:
         return fp.read()
 
+def update_versions(versions, release_matrix, release_version):
+    version_map = {
+        "preview": "preview",
+    }
```

Comment:
It looks like your version map will always be one entry `{ version : version}`. In that case, why does it need to be a map at all?